src/open_samus_returns_rando/specific_patches/chozo_seal_patches.py

```diff
@@ -45,5 +45,270 @@ def patch_dna_check(editor: PatcherEditor):
+        ]),
+    })
+
+    scenario = "s100_area10"
+    area10 = editor.get_scenario(scenario)
+    ap = "LE_RandoDNA_002"
+    ap_platform = "LE_Platform_RandoDNA_002"
+
+    accesspoint = _get_accesspoint(editor)
+    accesspoint_platform = _get_accesspoint_platform(editor)
+
+    editor.copy_actor(
+        scenario, (-4100.0, 11200.0, 10.0), accesspoint, ap, 16,
+    )
+
+    editor.copy_actor(
+        scenario, (-4100.0, 11200.0, 0.0), accesspoint_platform, ap_platform, 10,
+    )
+
+    area10.add_actor_to_entity_groups("collision_camera_022", ap)
+    area10.add_actor_to_entity_groups("collision_camera_024", ap)
+    area10.add_actor_to_entity_groups("collision_camera_022", ap_platform)
+    area10.add_actor_to_entity_groups("collision_camera_024", ap_platform)
+
+    area10.raw.actors[10][ap_platform]["components"][0]["arguments"][4]["value"] = ap
+
+    area10_map = editor.get_file("gui/minimaps/c10_samus/s100_area10.bmsmsd", Bmsmsd)
+    area10_map.raw["tiles"][253]["icons"] = ListContainer([CHOZO_SEAL_ICON])
+
+
+def add_surfaceb_seal(editor: PatcherEditor):
+    CHOZO_SEAL_ICON = Container({
+        "actor_name": "LE_RandoDNA",
+        "clear_condition": "",
+        "icon": "systemmechdna",
+        "icon_priority": 4,
+        "coordinates": ListContainer([
+            -28150.0, 300.0, 0.0
+        ]),
+    })
+
+    scenario = "s110_surfaceb"
+    surfaceb = editor.get_scenario(scenario)
+    ap = "LE_RandoDNA"
+    ap_platform = "LE_Platform_RandoDNA"
+
+    accesspoint = _get_accesspoint(editor)
+    accesspoint_platform = _get_accesspoint_platform(editor)
+
+    editor.copy_actor(
+        scenario, (-28150.0, 300.0, 0.0), accesspoint, ap, 16,
+    )
+
+    editor.copy_actor(
+        scenario, (-28150.0, 300.0, 0.0), accesspoint_platform, ap_platform, 10,
+    )
+
+    surfaceb.add_actor_to_entity_groups("collision_camera_017", ap)
+    surfaceb.add_actor_to_entity_groups("collision_camera_017", ap_platform)
+
+    surfaceb.raw.actors[10][ap_platform]["components"][0]["arguments"][4]["value"] = ap
+
+    surface_map = editor.get_file("gui/minimaps/c10_samus/s000_surface.bmsmsd", Bmsmsd)
+    surface_map.raw["tiles"][145]["icons"] = ListContainer([CHOZO_SEAL_ICON])
+
+
+def ensure_dependencies(editor: PatcherEditor) -> None:
+    SCENARIOS = [
+        "s033_area3b",
+        "s050_area5",
+        "s065_area6b",
+        "s100_area10",
+        "s110_surfaceb",
+    ]
+    for scenario in SCENARIOS:
+        editor.ensure_present_in_scenario(scenario, "maps/textures/dnaemptyfx_d.bctex")
+        for asset in editor.get_asset_names_in_folder("actors/props/systemmechdna"):
+            editor.ensure_present_in_scenario(scenario, asset)
+
+
 def patch_chozo_seals(editor: PatcherEditor):
     patch_dna_check(editor)
+    ensure_dependencies(editor)
+    add_area3b_seal(editor)
+    add_area5_seal(editor)
+    add_area6b_seal(editor)
+    add_area10_1_seal(editor)
+    add_area10_2_seal(editor)
+    add_surfaceb_seal(editor)
```

Comment:
Please don't matter endless copies of the same function, if all you really want to do is the same just with different parameters.

Reply:
The rewrite is completly missing my point here about `do [...] the same just with different parameters`.  It has swapped endless copies of a function for a new spaghetti function which checks for the scenario every second line.

You already started to use a proper data structure by extracting the coordinates but then sadly stopped.

I was thinking about something like the following. Such that if you want to modify, remove or add another seal, you don't have to touch any code besides adding a `NewChozoSeal` object to the list.

```python
import typing

class NewChozoSeal(typing.NamedTuple):
    scenario: str
    coordinates: tuple[(float, float, float)]
    ap_coordinates: tuple[(float, float, float)]
    tile_index: int
    entity_groups: list[str]
    ap_name: str = "LE_RandoDNA"
    platform_name: str = "LE_Platform_RandoDNA"

new_seals = [
    NewChozoSeal(
        "s033_area3b", [-3800.0, -16800.0, 0.0], [-3800.0, -16800.0, 0.0], 22, ["collision_camera_033", "PostGamma_004"]
    )
]

def add_chozo_seals(editor: PatcherEditor, new_seal: NewChozoSeal):
    template_ap = editor.get_scenario("s000_surface").raw.actors[16]["LE_ChozoUnlockAreaDNA"]
    template_platform = editor.get_scenario("s000_surface").raw.actors[10]["LE_Platform_ChozoUnlockAreaDNA"]
    
    CHOZO_SEAL_ICON = Container({
        "actor_name": "LE_RandoDNA",
        "clear_condition": "",
        "icon": "systemmechdna",
        "icon_priority": 4,
        "coordinates": ListContainer(new_seal.coordinates),
    })
    ...
    rest of the code
    ...
   # e.g. the whole entity group part will just look like this
    for entity_group in new_seal.entity_groups:
        scenario_file.add_actor_to_entity_groups(entity_group, new_ap, True)
        scenario_file.add_actor_to_entity_groups(entity_group, new_platform, True)

def patch_chozo_seals(editor: PatcherEditor):
    patch_dna_check(editor)
    for new_seal in new_seals:
        add_chozo_seals(editor, new_seal)
```